File: scripts/process_backend/_02/pb03/python/write_file.py
import sys
import time

lines = int(sys.argv[1]) if len(sys.argv) > 1 else 10000
filepath = sys.argv[2] if len(sys.argv) > 2 else 'output.csv'

start = time.time()
with open(filepath, 'w') as f:
    for i in range(lines):
        f.write(f"Line {i+1},Some data here\n")
end = time.time()

print(f"[Python] Wrote {lines} lines to {filepath} in {end - start:.4f} seconds.")

File: scripts/process_backend/_02/pb03/node/write_file.js
const fs = require('fs');
const lines = parseInt(process.argv[2]) || 10000;
const file = process.argv[3] || 'output.csv';

const start = Date.now();
const stream = fs.createWriteStream(file);
for (let i = 1; i <= lines; i++) {
    stream.write(`Line ${i},Some data here\n`);
}
stream.end(() => {
    const end = Date.now();
    console.log(`[NodeJS] Wrote ${lines} lines to ${file} in ${(end - start)/1000} seconds.`);
});